library/alloc/src/raw_vec.rs

```diff
@@ -394,6 +394,28 @@ impl<T, A: Allocator> RawVec<T, A> {
         let cap = cmp::max(self.cap * 2, required_cap);
         let cap = cmp::max(Self::MIN_NON_ZERO_CAP, cap);
 
+        let alloc_size = cap.checked_mul(mem::align_of::<T>()).ok_or(CapacityOverflow)?;
```

Comment:
This incorrect, which became clear when `[u8; n]` tests started to run. I'll fix to use `size_of`. Please someone correct me if `size_of` does not already include alignment requirements.

Reply:
Rust guarantees that size is a multiple of align.